### DISCUSSION TITLE
How to use styled if using the babel transform?

### DISCUSSION THREAD

**Discussion by jmeistrich** [nativewind/nativewind#70] [Q&A] [answered]
It seems like the babel transform always creates a `StyledComponent` any time it sees `className`, so does that mean we should never use `styled` manually? There are cases where I would like to use it to adding multiple styling properties. It does seem to work, but from reading the code it looks like that would wrap it in `StyledComponent` twice?

**Comment by marklawlor** [comment, marked as answer]
> so does that mean we should never use styled manually

My current vision for the library is that projects using the Babel transform will be using a mix of `<StyledComponent />` and `styled()`.

The styled() options make it a useful component factory and can save your a lot of boilerplate in your code. So I would envision that many of the low level components (eg typography, icons, etc) are written with `styled()` and the rest are using the transform.

> looks like that would wrap it in StyledComponent twice?

It does. If you use `styled()` and the babel transform it will produce a component that is `styled(styled(component))`. 

This does increase the React tree depth, but it's a working solution for now.

I have plans to optimise it when I start focusing on performance. So if you write `styled(styled(component, optionsA), optionsB)` its actually executed as `styled(component, { ...optionsA, ...optionsB })`.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
